[CI] ReservedRealmElasticAutoconfigIntegTests testAutoconfigFailedPasswordPromotion failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #6404 / windows-2019_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/frnib3rje4rfa)
- [elasticsearch-pull-request #56520 / part-4](https://gradle-enterprise.elastic.co/s/qwo4hbdhbyrck)
- [elasticsearch-periodic-platform-support #6292 / windows-2022_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/amkfvjyoljryq)

**Reproduction Line:**
```
gradlew ":x-pack:plugin:security:internalClusterTest" --tests "org.elasticsearch.xpack.security.authc.esnative.ReservedRealmElasticAutoconfigIntegTests.testAutoconfigFailedPasswordPromotion" -Dtests.seed=A302213C687EA937 -Dtests.locale=sv-AX -Dtests.timezone=Antarctica/Rothera -Druntime.java=23
```

**Applicable branches:**
8.18

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.security.authc.esnative.ReservedRealmElasticAutoconfigIntegTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testAutoconfigFailedPasswordPromotion),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: is <503>
     but: was <401>
```

**Issue Reasons:**
- [8.18] 3 failures in test testAutoconfigFailedPasswordPromotion (0.5% fail rate in 658 executions)
- [8.18] 2 failures in pipeline elasticsearch-periodic-platform-support (11.8% fail rate in 17 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -346,9 +346,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.StatelessHollowIndexShardsIT
   method: testStressWithoutFailures
   issue: https://github.com/elastic/elasticsearch/issues/148310
-- class: org.elasticsearch.xpack.security.authc.esnative.ReservedRealmElasticAutoconfigIntegTests
-  method: testAutoconfigFailedPasswordPromotion
-  issue: https://github.com/elastic/elasticsearch/issues/122668
 - class: org.elasticsearch.xpack.esql.analysis.VerifierTests
   method: testFlattenedSorting
   issue: https://github.com/elastic/elasticsearch/issues/148374

